<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>protein-interactions-(publications-fb5bc3bb0c5d40468da36d279a114a78 – GD benchmark sets</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-a202ed359b7240dcab911ed64cbd0efc.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body>

<div id="quarto-search-results"></div>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#protein-interactions-publications" id="toc-protein-interactions-publications" class="nav-link active" data-scroll-target="#protein-interactions-publications"><span class="header-section-number">1</span> Protein Interactions (Publications)</a></li>
  <li><a href="#body" id="toc-body" class="nav-link" data-scroll-target="#body"><span class="header-section-number">2</span> Body</a>
  <ul class="collapse">
  <li><a href="#protein-interaction-publications" id="toc-protein-interaction-publications" class="nav-link" data-scroll-target="#protein-interaction-publications"><span class="header-section-number">2.1</span> Protein Interaction Publications</a>
  <ul class="collapse">
  <li><a href="#statistics" id="toc-statistics" class="nav-link" data-scroll-target="#statistics">Statistics</a></li>
  <li><a href="#description-from-the-literature" id="toc-description-from-the-literature" class="nav-link" data-scroll-target="#description-from-the-literature">Description from the Literature</a></li>
  <li><a href="#example-figures" id="toc-example-figures" class="nav-link" data-scroll-target="#example-figures">Example Figures</a></li>
  </ul></li>
  <li><a href="#protein-homology-lost" id="toc-protein-homology-lost" class="nav-link" data-scroll-target="#protein-homology-lost"><span class="header-section-number">2.2</span> Protein Homology (Lost)</a>
  <ul class="collapse">
  <li><a href="#description-from-the-literature-1" id="toc-description-from-the-literature-1" class="nav-link" data-scroll-target="#description-from-the-literature-1">Description from the Literature</a></li>
  <li><a href="#example-figures-1" id="toc-example-figures-1" class="nav-link" data-scroll-target="#example-figures-1">Example Figures</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block"></header>




<section id="protein-interactions-publications" class="level1" data-number="1">
<h1 data-number="1"><span class="header-section-number">1</span> Protein Interactions (Publications)</h1>
<p>Note: Various types of protein interaction graphs are recorded in other network collections. We highlight two of them from the surveyed papers. First, we have the temporal Protein Interaction Publications, showing the history of how protein interactions were described in the literature. Second, the Protein Homology graph was provided by the Large Graph Layout project, their links are now broken and the dataset lost. Nonetheless, http://snap.stanford.edu/biodata/index.html and http://konect.cc/networks/ both also have various dataset of human/other species protein interactions. Origin Notes: Protein Interaction Publications were collected from the biological Pathway Commons Database: http://www.pathwaycommons.org/. graph features handled: Dynamic, Dynamic (discrete), Large, Multigraph Graph features in papers: generic,large,dynamic,dynamic (discrete),layered graphs,n-layers Origin Paper: LGL: Creating a Map of Protein Function with an Algorithm for Visualizing Very Large Biological Networks (https://www.notion.so/LGL-Creating-a-Map-of-Protein-Function-with-an-Algorithm-for-Visualizing-Very-Large-Biological-Netw-518c7160f2f24cd08c6bbb7b4f43216c?pvs=21), TimeArcs: Visualizing Fluctuations in Dynamic Networks (https://www.notion.so/TimeArcs-Visualizing-Fluctuations-in-Dynamic-Networks-e671c46ecfa444efaf28392636402266?pvs=21) Originally found at: https://github.com/CreativeCodingLab/TimeArcs/tree/master/IndexCards</p>
<p>https://lgl.sourceforge.net/ (download link broken) Size: 2961 nodes, 5267 edges Number of Graphs: 1 Appeared in years: 2008,2016 Type of Collection: Single Graph is it stored properly?: No must be analyzed: No In repo?: Yes Related to Literature - Algorithm (1) (Dataset tag relations): Rapid Graph Layout Using Space Filling Curves (https://www.notion.so/Rapid-Graph-Layout-Using-Space-Filling-Curves-847c46047b5c400bb9dcf339c8d42f12?pvs=21), TimeArcs: Visualizing Fluctuations in Dynamic Networks (https://www.notion.so/TimeArcs-Visualizing-Fluctuations-in-Dynamic-Networks-968889d3ca4a4109aca698513515e837?pvs=21) cleaned format?: Yes duplicate?: No link works?: No Added in paper: No OSF link json: https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d94abc0c2b4d0e8d38629c Origin paper plaintext: LGL: Creating a Map of Protein Function with an Algorithm for Visualizing Very Large Biological Networks, TimeArcs: Visualizing Fluctuations in Dynamic Networks Page id: fb5bc3bb0c5d40468da36d279a114a78 unavailable/skip: No Cleaned ALL data: No OSF link gexf: https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d9487494a6be101c12e7a5 OSF link gml: https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d96cd4803e0c0ca9558d2b OSF link graphml: https://files.osf.io/v1/resources/j7ucv/providers/osfstorage/64d971704cf748115a055882 first look: Yes sparkline data: {‘min’: 1, ‘max’: 227, ‘step_size’: 10, ‘num_bins’: 23, ‘bins’: [0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130, 140, 150, 160, 170, 180, 190, 200, 210, 220], ‘node_degree’: [2768, 106, 38, 17, 12, 7, 3, 2, 3, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1]} Related to Literature - Algorithm (Dataset tag relations) 1: Rapid Graph Layout Using Space Filling Curves (../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/Rapid%20Graph%20Layout%20Using%20Space%20Filling%20Curves%2010601cd6078a4ea18b17c7d40eda0041.md), TimeArcs: Visualizing Fluctuations in Dynamic Networks (../Benchmark%20sets%200cc6b5e454304aec98f3b59b1a720476/Literature%20ad87f14e7097454fb2f784e2c8a2797a/Literature%20-%20Algorithm%2012e01bfc60a84007aa7d2d34293e123d/TimeArcs%20Visualizing%20Fluctuations%20in%20Dynamic%20Netwo%209d27b7e02aec4b80bc15447255eb4f4c.md)</p>
</section>
<section id="body" class="level1" data-number="2">
<h1 data-number="2"><span class="header-section-number">2</span> Body</h1>
<section id="protein-interaction-publications" class="level2" data-number="2.1">
<h2 data-number="2.1" class="anchored" data-anchor-id="protein-interaction-publications"><span class="header-section-number">2.1</span> Protein Interaction Publications</h2>
<section id="statistics" class="level3">
<h3 class="anchored" data-anchor-id="statistics">Statistics</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Protein Interactions (Publications) fb5bc3bb0c5d40468da36d279a114a78/degree_distr.svg" class="img-fluid figure-img"></p>
<figcaption>degree_distr.svg</figcaption>
</figure>
</div>
</section>
<section id="description-from-the-literature" class="level3">
<h3 class="anchored" data-anchor-id="description-from-the-literature">Description from the Literature</h3>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.12882">TimeArcs: Visualizing Fluctuations in Dynamic Networks</a>:</p>
<blockquote class="blockquote">
<p>The data contains the publication information (such as publication year, author, and textual evidence) of interactions between pairs of proteins, as well as their specific interaction type.</p>
</blockquote>
<blockquote class="blockquote">
<p>When there are multiple arcs connecting two proteins, it falls into one of the two circumstances. If they have the same color, these arcs indicate that there are supporting evidences in different publications which&nbsp;<em>confirm</em>&nbsp;the interaction between two elements. On the other hand, if they have the different colors, the more recent appearance provides either more detailed knowledge about the interaction or shows a&nbsp;<em>conflict</em>&nbsp;between different articles regarding the way in which these proteins interact.</p>
</blockquote>
</section>
<section id="example-figures" class="level3">
<h3 class="anchored" data-anchor-id="example-figures">Example Figures</h3>
<p>From <a href="https://onlinelibrary.wiley.com/doi/10.1111/cgf.12882">TimeArcs: Visualizing Fluctuations in Dynamic Networks</a>:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Protein Interactions (Publications) fb5bc3bb0c5d40468da36d279a114a78/Untitled.png" class="img-fluid figure-img"></p>
<figcaption>Untitled</figcaption>
</figure>
</div>
<p><strong>Fig. 10</strong>: TimeArcs visualization for interactions around PCAF protein. (1), (2), and (3) in the figure are supporting evidences in literature of “PCAF binds MAML”.</p>
</section>
</section>
<section id="protein-homology-lost" class="level2" data-number="2.2">
<h2 data-number="2.2" class="anchored" data-anchor-id="protein-homology-lost"><span class="header-section-number">2.2</span> Protein Homology (Lost)</h2>
<section id="description-from-the-literature-1" class="level3">
<h3 class="anchored" data-anchor-id="description-from-the-literature-1">Description from the Literature</h3>
<p>From [Rapid Graph Layout Using Space Filling Curves:](https://ieeexplore.ieee.org/document/4658143)</p>
<blockquote class="blockquote">
<p>The “pgraph” dataset (shown in&nbsp;Figures 2&nbsp;and&nbsp;6) is a protein homology graph, which is relatively dense&nbsp;[7].</p>
</blockquote>
<p>Consists of 28, 854 vertices and 1,180, 816 edges, found in Table 1 of paper above.</p>
</section>
<section id="example-figures-1" class="level3">
<h3 class="anchored" data-anchor-id="example-figures-1">Example Figures</h3>
<p>From <a href="https://ieeexplore.ieee.org/document/4658143">Rapid Graph Layout Using Space Filling Curves:</a></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Protein Interactions (Publications) fb5bc3bb0c5d40468da36d279a114a78/Untitled 1.png" class="img-fluid figure-img"></p>
<figcaption>Untitled</figcaption>
</figure>
</div>
<p><strong>Fig. 2.</strong> <em>A protein homology graph laid out with our space filling curve based approach.</em>&nbsp;Color corresponds to depth in the clustering hierarchy. |<em>V</em>| = 28, 854,|<em>E</em>| = 1, 180, 816</p>
<p>From <a href="https://www.sciencedirect.com/science/article/pii/S0022283604004851">LGL: Creating a Map of Protein Function with an Algorithm for Visualizing Very Large Biological Networks</a></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Protein Interactions (Publications) fb5bc3bb0c5d40468da36d279a114a78/Untitled 2.png" class="img-fluid figure-img"></p>
<figcaption>Untitled</figcaption>
</figure>
</div>
<p><strong>Fig. 3.</strong>&nbsp;The complete protein homology map. A layout of the entire protein homology map; a total of 11,516 connected sets containing 111,604 proteins (vertices) with 1,912,684 edges. The largest connected set is shown more clearly in the inset and is enlarged further in&nbsp;<a href="https://www.sciencedirect.com/science/article/pii/S0022283604004851?via%3Dihub#FIG4">Figure 4</a>.</p>
<p>== STOP RENDERING == broken link referenced in <a href="../Benchmark sets 0cc6b5e454304aec98f3b59b1a720476/Literature ad87f14e7097454fb2f784e2c8a2797a/Literature - Algorithm 12e01bfc60a84007aa7d2d34293e123d/Rapid Graph Layout Using Space Filling Curves 10601cd6078a4ea18b17c7d40eda0041.md"><strong>Rapid Graph Layout Using Space Filling Curves</strong></a> :</p>
<p>Same link referenced in original paper,</p>
<p>I also found <a href="https://lgl.sourceforge.net/">https://lgl.sourceforge.net/</a> which contains the link to supposedly download the dataset but it is also broken</p>
<p>Reached out to author - Alex Adai</p>
<p><a href="http://www.marcottelab.org/index.php/OldBioinformaticsServerlgl/"></a></p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
});
</script>
</div> <!-- /content -->




</body></html>